In [1]:
import pandas as pd
import numpy as np

census_designated_places = pd.read_csv('PLACES_2022_census_designated_places.csv')
census_tract = pd.read_csv('PLACES_2022_census_tract.csv')
county = pd.read_csv('PLACES_2022_county.csv')
zcta = pd.read_csv('PLACES_2022_zcta.csv')

/var/folders/73/p4smzwgd1txgby5s7wq0gt700000gn/T/ipykernel_34784/1291202202.py:7: DtypeWarning: Columns (8,9) have mixed types. Specify dtype option on import or set low_memory=False.
  zcta = pd.read_csv('PLACES_2022_zcta.csv')


In [2]:
print(census_designated_places.columns, census_tract.columns, county.columns, zcta.columns)

Index(['Year', 'StateAbbr', 'StateDesc', 'LocationName', 'DataSource',
       'Category', 'Measure', 'Data_Value_Unit', 'Data_Value_Type',
       'Data_Value', 'Data_Value_Footnote_Symbol', 'Data_Value_Footnote',
       'Low_Confidence_Limit', 'High_Confidence_Limit', 'TotalPopulation',
       'Geolocation', 'LocationID', 'CategoryID', 'MeasureId',
       'DataValueTypeID', 'Short_Question_Text', 'States', 'Counties'],
      dtype='object') Index(['Year', 'StateAbbr', 'StateDesc', 'CountyName', 'CountyFIPS',
       'LocationName', 'DataSource', 'Category', 'Measure', 'Data_Value_Unit',
       'Data_Value_Type', 'Data_Value', 'Data_Value_Footnote_Symbol',
       'Data_Value_Footnote', 'Low_Confidence_Limit', 'High_Confidence_Limit',
       'TotalPopulation', 'Geolocation', 'LocationID', 'CategoryID',
       'MeasureId', 'DataValueTypeID', 'Short_Question_Text', 'States',
       'Counties'],
      dtype='object') Index(['Year', 'StateAbbr', 'StateDesc', 'LocationName', 'DataSource',
    

In [3]:
print(len(census_designated_places), len(census_tract), len(county), len(zcta))

1704502 2161543 188456 969023


## 'census_designated_places' and steps

In [4]:
census_designated_places.head()

Year StateAbbr StateDesc    LocationName DataSource         Category  \
0  2020        AL   Alabama     Albertville      BRFSS  Health Outcomes   
1  2020        AL   Alabama  Alexander City      BRFSS  Health Outcomes   
2  2020        AL   Alabama      Alexandria      BRFSS  Health Outcomes   
3  2020        AL   Alabama      Aliceville      BRFSS  Health Outcomes   
4  2019        AL   Alabama       Andalusia      BRFSS       Prevention   

                                             Measure Data_Value_Unit  \
0                Stroke among adults aged >=18 years               %   
1                Stroke among adults aged >=18 years               %   
2            Depression among adults aged >=18 years               %   
3  Cancer (excluding skin cancer) among adults ag...               %   
4  Cholesterol screening among adults aged >=18 y...               %   

           Data_Value_Type  Data_Value  ... High_Confidence_Limit  \
0  Age-adjusted prevalence         3.6  ...                   3.8   
1         Crude prevalence         4.6  ...                   4.9   
2         Crude prevalence        26.2  ...                  27.6   
3         Crude prevalence         7.0  ...                   7.3   
4  Age-adjusted prevalence        86.0  ...                  86.5   

  TotalPopulation                       Geolocation  LocationID  CategoryID  \
0           21282  POINT (-86.21214093 34.26441274)      100988     HLTHOUT   
1           14972  POINT (-85.93972723 32.93023222)      101132     HLTHOUT   
2            3917  POINT (-85.87920682 33.76747698)      101180     HLTHOUT   
3            2486  POINT (-88.15202475 33.12955557)      101228     HLTHOUT   
4            9015  POINT (-86.47806878 31.31012814)      101708     PREVENT   

    MeasureId  DataValueTypeID    Short_Question_Text States Counties  
0      STROKE        AgeAdjPrv                 Stroke   29.0   1590.0  
1      STROKE           CrdPrv                 Stroke   29.0   1638.0  
2  DEPRESSION           CrdPrv             Depression   29.0   1491.0  
3      CANCER           CrdPrv   Cancer (except skin)   29.0   1632.0  
4  CHOLSCREEN        AgeAdjPrv  Cholesterol Screening   29.0   1539.0  

[5 rows x 23 columns]

In [5]:
census_designated_places.isnull().sum()

Year                                0
StateAbbr                           0
StateDesc                           0
LocationName                        0
DataSource                          0
Category                            0
Measure                             0
Data_Value_Unit                     0
Data_Value_Type                     0
Data_Value                      31471
Data_Value_Footnote_Symbol    1673031
Data_Value_Footnote           1673031
Low_Confidence_Limit            31471
High_Confidence_Limit           31471
TotalPopulation                     0
Geolocation                         0
LocationID                          0
CategoryID                          0
MeasureId                           0
DataValueTypeID                     0
Short_Question_Text                 0
States                          22986
Counties                         3284
dtype: int64

In [6]:
print(census_designated_places.Data_Value_Footnote_Symbol.unique())
print(census_designated_places.Data_Value_Footnote.unique())
print(census_designated_places.DataSource.unique())

[nan '#']
[nan
 'Not estimated due to lack of population sizes at particular age groups']
['BRFSS']


In [7]:
print(census_designated_places.CategoryID.unique())
print(census_designated_places.MeasureId.unique())
print(census_designated_places.DataValueTypeID.unique()) # could be replaced with binary

['HLTHOUT' 'PREVENT' 'RISKBEH' 'HLTHSTAT']
['STROKE' 'DEPRESSION' 'CANCER' 'CHOLSCREEN' 'DIABETES' 'CHD' 'COPD'
 'ARTHRITIS' 'CASTHMA' 'CHECKUP' 'CERVICAL' 'OBESITY' 'KIDNEY' 'BPHIGH'
 'BINGE' 'TEETHLOST' 'ACCESS2' 'COLON_SCREEN' 'CSMOKING' 'MAMMOUSE'
 'SLEEP' 'PHLTH' 'MHLTH' 'LPA' 'COREM' 'GHLTH' 'COREW' 'DENTAL' 'BPMED'
 'HIGHCHOL']
['AgeAdjPrv' 'CrdPrv']


## Decide to just drop the ~30,000 missing data rows, as it is a small compared to total # of rows being 1704502, however, may want to try other methods moving forward (spend more time seeing why these rows are missing data values)

In [8]:
del census_designated_places['StateDesc'] # repitition of previous column
del census_designated_places['Data_Value_Footnote_Symbol'] # all nan
del census_designated_places['Data_Value_Unit'] # everything is % in this column
del census_designated_places['Data_Value_Footnote'] # all nan
del census_designated_places['DataSource'] # only 1 value
del census_designated_places['Data_Value_Type'] # DataValueTypeID columns same but provides simplification
del census_designated_places['States'] # repeat data, also not even included in the table pdf they gave
del census_designated_places['Counties'] # repeat data, also not even included in the table pdf they gave
del census_designated_places['LocationID'] # not helpful
del census_designated_places['Short_Question_Text'] # repeat of measureID
del census_designated_places['Category'] # repeat in categoryID
del census_designated_places['Measure'] # measureID summarizes it

census_designated_places['ConfidenceLimit'] = census_designated_places[['Low_Confidence_Limit', 'High_Confidence_Limit']].apply(tuple, axis=1)
del census_designated_places['Low_Confidence_Limit']
del census_designated_places['High_Confidence_Limit']

census_designated_places = census_designated_places[census_designated_places['Data_Value'].notna()]
census_designated_places = census_designated_places.rename(columns={'StateAbbr': 'State', 'Data_Value': 'DataValue', 'Short_Question_Text':'ShortQuestionText', 'MeasureId':'MeasureID', 'Geolocation':'GeoLocation'})

census_designated_places.isnull().sum() # now left with no NA values

Year               0
State              0
LocationName       0
DataValue          0
TotalPopulation    0
GeoLocation        0
CategoryID         0
MeasureID          0
DataValueTypeID    0
ConfidenceLimit    0
dtype: int64

In [9]:
census_designated_places

Year State    LocationName  DataValue  TotalPopulation  \
0        2020    AL     Albertville        3.6            21282   
1        2020    AL  Alexander City        4.6            14972   
2        2020    AL      Alexandria       26.2             3917   
3        2020    AL      Aliceville        7.0             2486   
4        2019    AL       Andalusia       86.0             9015   
...       ...   ...             ...        ...              ...   
1704497  2020    WY        Riverton       31.3            10736   
1704498  2020    WY        Shoshoni       37.5              649   
1704499  2020    WY          Wright       39.0             1807   
1704500  2020    WY         Ralston       34.3              280   
1704501  2020    WY     Thermopolis       29.6             3009   

                              GeoLocation CategoryID   MeasureID  \
0        POINT (-86.21214093 34.26441274)    HLTHOUT      STROKE   
1        POINT (-85.93972723 32.93023222)    HLTHOUT      STROKE   
2        POINT (-85.87920682 33.76747698)    HLTHOUT  DEPRESSION   
3        POINT (-88.15202475 33.12955557)    HLTHOUT      CANCER   
4        POINT (-86.47806878 31.31012814)    PREVENT  CHOLSCREEN   
...                                   ...        ...         ...   
1704497  POINT (-108.4240535 43.03136645)    RISKBEH       SLEEP   
1704498  POINT (-108.1046161 43.23814686)    PREVENT       COREM   
1704499  POINT (-105.4959023 43.74853082)    PREVENT       COREM   
1704500  POINT (-108.8823081 44.70849191)    PREVENT       COREW   
1704501  POINT (-108.2139638 43.64787387)    PREVENT       COREW   

        DataValueTypeID ConfidenceLimit  
0             AgeAdjPrv      (3.4, 3.8)  
1                CrdPrv      (4.3, 4.9)  
2                CrdPrv    (24.9, 27.6)  
3                CrdPrv      (6.7, 7.3)  
4             AgeAdjPrv    (85.4, 86.5)  
...                 ...             ...  
1704497          CrdPrv    (30.6, 32.0)  
1704498       AgeAdjPrv    (30.3, 45.2)  
1704499          CrdPrv    (29.5, 48.5)  
1704500       AgeAdjPrv    (29.5, 39.4)  
1704501          CrdPrv    (23.7, 35.8)  

[1673031 rows x 10 columns]

## Repeating operations but for 'census_tract' df

In [10]:
del census_tract['StateDesc'] # repitition of previous column
del census_tract['Data_Value_Footnote_Symbol'] # all nan
del census_tract['Data_Value_Unit'] # everything is % in this column
del census_tract['Data_Value_Footnote'] # all nan
del census_tract['DataSource'] # only 1 value
del census_tract['Data_Value_Type'] # DataValueTypeID columns same but provides simplification
del census_tract['States'] # repeat data, also not even included in the table pdf they gave
del census_tract['Counties'] # repeat data, also not even included in the table pdf they gave
del census_tract['LocationID'] # not helpful
del census_tract['Short_Question_Text'] # repeat of measureID
del census_tract['Category'] # repeat in categoryID
del census_tract['Measure'] # measureID summarizes it

del census_tract['CountyFIPS']

census_tract['ConfidenceLimit'] = census_tract[['Low_Confidence_Limit', 'High_Confidence_Limit']].apply(tuple, axis=1)
del census_tract['Low_Confidence_Limit']
del census_tract['High_Confidence_Limit']

census_tract = census_tract[census_tract['Data_Value'].notna()]
census_tract = census_tract.rename(columns={'StateAbbr': 'State', 'Data_Value': 'DataValue', 'Short_Question_Text':'ShortQuestionText', 'MeasureId':'MeasureID', 'Geolocation':'GeoLocation'})

census_tract.isnull().sum() # now left with no NA values

Year               0
State              0
CountyName         0
LocationName       0
DataValue          0
TotalPopulation    0
GeoLocation        0
CategoryID         0
MeasureID          0
DataValueTypeID    0
ConfidenceLimit    0
dtype: int64

In [11]:
census_tract

Year State CountyName  LocationName  DataValue  TotalPopulation  \
0        2020    AL    Autauga    1001020200        7.6             2170   
1        2020    AL    Autauga    1001020400       33.2             4386   
2        2020    AL    Autauga    1001021000       40.3             2894   
3        2020    AL    Baldwin    1003010905        2.9             8011   
4        2020    AL    Baldwin    1003010906       27.6             4302   
...       ...   ...        ...           ...        ...              ...   
2161538  2020    WI   Waukesha   55133204002       62.0             5479   
2161539  2020    WI       Rock   55105002700       10.6             2571   
2161540  2020    WI   Waukesha   55133203200        7.4             4106   
2161541  2020    WI   Waukesha   55133203101       77.8             6536   
2161542  2020    WI   Waukesha   55133200400        2.3             6582   

                              GeoLocation CategoryID     MeasureID  \
0        POINT (-86.47338874 32.47425254)    HLTHOUT          COPD   
1        POINT (-86.44370417 32.47203959)    HLTHOUT       OBESITY   
2        POINT (-86.75607371 32.60894544)    HLTHOUT       OBESITY   
3        POINT (-87.75682132 30.53592075)    HLTHOUT        STROKE   
4        POINT (-87.68753351 30.56418914)    HLTHOUT    DEPRESSION   
...                                   ...        ...           ...   
2161538   POINT (-88.4824858 42.88723077)    PREVENT  COLON_SCREEN   
2161539    POINT (-89.24626648 42.548431)   HLTHSTAT         PHLTH   
2161540  POINT (-88.31331246 43.07069963)    HLTHOUT        CANCER   
2161541  POINT (-88.27895206 43.04127835)    PREVENT        DENTAL   
2161542  POINT (-88.10191212 43.13413751)    HLTHOUT        KIDNEY   

        DataValueTypeID ConfidenceLimit  
0                CrdPrv      (6.8, 8.5)  
1                CrdPrv    (31.5, 35.0)  
2                CrdPrv    (39.3, 41.2)  
3                CrdPrv      (2.7, 3.2)  
4                CrdPrv    (26.3, 28.9)  
...                 ...             ...  
2161538          CrdPrv    (59.0, 64.9)  
2161539          CrdPrv     (9.6, 11.6)  
2161540          CrdPrv      (7.0, 7.8)  
2161541          CrdPrv    (74.7, 80.6)  
2161542          CrdPrv      (2.1, 2.5)  

[2161542 rows x 11 columns]

## Repeating operations but for 'zcta' df

In [12]:
zcta

Year  LocationName DataSource               Category  \
0       2020          1005      BRFSS        Health Outcomes   
1       2020          1007      BRFSS        Health Outcomes   
2       2020          1008      BRFSS        Health Outcomes   
3       2020          1009      BRFSS        Health Outcomes   
4       2020          1026      BRFSS        Health Outcomes   
...      ...           ...        ...                    ...   
969018  2020         99138      BRFSS             Prevention   
969019  2020         99703      BRFSS  Health Risk Behaviors   
969020  2019         99701      BRFSS        Health Outcomes   
969021  2019         99338      BRFSS             Prevention   
969022  2020         99643      BRFSS             Prevention   

                                                  Measure Data_Value_Unit  \
0                  Arthritis among adults aged >=18 years               %   
1                     Stroke among adults aged >=18 years               %   
2                    Obesity among adults aged >=18 years               %   
3                    Obesity among adults aged >=18 years               %   
4       Chronic kidney disease among adults aged >=18 ...               %   
...                                                   ...             ...   
969018  Fecal occult blood test, sigmoidoscopy, or col...               %   
969019  No leisure-time physical activity among adults...               %   
969020   High blood pressure among adults aged >=18 years               %   
969021  Cholesterol screening among adults aged >=18 y...               %   
969022  Cervical cancer screening among adult women ag...               %   

         Data_Value_Type  Data_Value Data_Value_Footnote_Symbol  \
0       Crude prevalence        27.2                        NaN   
1       Crude prevalence         2.1                        NaN   
2       Crude prevalence        29.2                        NaN   
3       Crude prevalence        29.7                        NaN   
4       Crude prevalence         2.8                        NaN   
...                  ...         ...                        ...   
969018  Crude prevalence        60.3                        NaN   
969019  Crude prevalence        17.5                        NaN   
969020  Crude prevalence        30.9                        NaN   
969021  Crude prevalence        90.3                        NaN   
969022  Crude prevalence        77.8                        NaN   

       Data_Value_Footnote  ...  High_Confidence_Limit  TotalPopulation  \
0                      NaN  ...                   28.6             5079   
1                      NaN  ...                    2.4            14649   
2                      NaN  ...                   30.5             1263   
3                      NaN  ...                   31.5              741   
4                      NaN  ...                    3.0              946   
...                    ...  ...                    ...              ...   
969018                 NaN  ...                   62.2             1209   
969019                 NaN  ...                   19.9             7065   
969020                 NaN  ...                   31.3            19019   
969021                 NaN  ...                   90.9            11555   
969022                 NaN  ...                   81.0              218   

                             Geolocation LocationID  CategoryID     MeasureId  \
0       POINT (-72.10611705 42.42009493)       1005     HLTHOUT     ARTHRITIS   
1       POINT (-72.40031647 42.27869142)       1007     HLTHOUT        STROKE   
2       POINT (-72.95202996 42.18429555)       1008     HLTHOUT       OBESITY   
3       POINT (-72.33968366 42.20953238)       1009     HLTHOUT       OBESITY   
4        POINT (-72.92024618 42.4633281)       1026     HLTHOUT        KIDNEY   
...                                  ...        ...         ...           ...   
969018  POINT (-118.3787355 48.22230839) 

In [13]:
del zcta['Data_Value_Footnote_Symbol'] # all nan
del zcta['Data_Value_Unit'] # everything is % in this column
del zcta['Data_Value_Footnote'] # all nan
del zcta['DataSource'] # only 1 value
del zcta['Data_Value_Type'] # DataValueTypeID columns same but provides simplification

# OPT NOT TO DELETE STATES AND COUNTIES IN THIS CASE

#del zcta['States'] # repeat data, also not even included in the table pdf they gave
#del zcta['Counties'] # repeat data, also not even included in the table pdf they gave

del zcta['LocationID'] # not helpful
del zcta['Short_Question_Text'] # repeat of measureID
del zcta['Category'] # repeat in categoryID
del zcta['Measure'] # measureID summarizes it

zcta['ConfidenceLimit'] = zcta[['Low_Confidence_Limit', 'High_Confidence_Limit']].apply(tuple, axis=1)
del zcta['Low_Confidence_Limit']
del zcta['High_Confidence_Limit']

zcta = zcta[zcta['Data_Value'].notna()]
zcta = zcta[zcta['States'].notna()]
zcta = zcta[zcta['Counties'].notna()] # dropping 870 rows
zcta = zcta.rename(columns={'StateAbbr': 'State', 'Data_Value': 'DataValue', 'Short_Question_Text':'ShortQuestionText', 'MeasureId':'MeasureID', 'Geolocation':'GeoLocation'})

zcta.isnull().sum() # now left with no NA values

Year               0
LocationName       0
DataValue          0
TotalPopulation    0
GeoLocation        0
CategoryID         0
MeasureID          0
DataValueTypeID    0
States             0
Counties           0
ConfidenceLimit    0
dtype: int64

In [14]:
zcta

Year  LocationName  DataValue  TotalPopulation  \
0       2020          1005       27.2             5079   
1       2020          1007        2.1            14649   
2       2020          1008       29.2             1263   
3       2020          1009       29.7              741   
4       2020          1026        2.8              946   
...      ...           ...        ...              ...   
969018  2020         99138       60.3             1209   
969019  2020         99703       17.5             7065   
969020  2019         99701       30.9            19019   
969021  2019         99338       90.3            11555   
969022  2020         99643       77.8              218   

                             GeoLocation CategoryID     MeasureID  \
0       POINT (-72.10611705 42.42009493)    HLTHOUT     ARTHRITIS   
1       POINT (-72.40031647 42.27869142)    HLTHOUT        STROKE   
2       POINT (-72.95202996 42.18429555)    HLTHOUT       OBESITY   
3       POINT (-72.33968366 42.20953238)    HLTHOUT       OBESITY   
4        POINT (-72.92024618 42.4633281)    HLTHOUT        KIDNEY   
...                                  ...        ...           ...   
969018  POINT (-118.3787355 48.22230839)    PREVENT  COLON_SCREEN   
969019  POINT (-147.6232707 64.83280824)    RISKBEH           LPA   
969020  POINT (-149.1566777 67.48759306)    HLTHOUT        BPHIGH   
969021   POINT (-119.2591805 46.1316345)    PREVENT    CHOLSCREEN   
969022  POINT (-153.3435631 57.21786791)    PREVENT      CERVICAL   

       DataValueTypeID  States  Counties ConfidenceLimit  
0               CrdPrv    25.0    1919.0    (25.8, 28.6)  
1               CrdPrv    25.0    1914.0      (1.9, 2.4)  
2               CrdPrv    25.0    1913.0    (27.9, 30.5)  
3               CrdPrv    25.0    1913.0    (28.0, 31.5)  
4               CrdPrv    25.0    1914.0      (2.6, 3.0)  
...                ...     ...       ...             ...  
969018          CrdPrv     6.0    2980.0    (58.4, 62.2)  
969019          CrdPrv    51.0    1673.0    (15.3, 19.9)  
969020          CrdPrv    51.0       4.0    (30.4, 31.3)  
969021          CrdPrv     6.0    2955.0    (89.6, 90.9)  
969022          CrdPrv    51.0     112.0    (74.5, 81.0)  

[957871 rows x 11 columns]

# NOTE: 'county' dataset is slightly different than the others

## Repeating operations but for 'county' df (drop an additional 60 rows that don't have location)

In [15]:
county

Year StateAbbr      StateDesc LocationName DataSource  \
0       2020        US  United States          NaN      BRFSS   
1       2020        AL        Alabama      Colbert      BRFSS   
2       2020        AL        Alabama        Coosa      BRFSS   
3       2019        AL        Alabama         Dale      BRFSS   
4       2020        AL        Alabama         Dale      BRFSS   
...      ...       ...            ...          ...        ...   
188451  2019        WA     Washington       Skagit      BRFSS   
188452  2020        WI      Wisconsin       Oneida      BRFSS   
188453  2020        WI      Wisconsin       Oconto      BRFSS   
188454  2020        WV  West Virginia      Kanawha      BRFSS   
188455  2020        WI      Wisconsin  Fond du Lac      BRFSS   

                     Category  \
0                  Prevention   
1             Health Outcomes   
2             Health Outcomes   
3                  Prevention   
4             Health Outcomes   
...                       ...   
188451        Health Outcomes   
188452          Health Status   
188453  Health Risk Behaviors   
188454          Health Status   
188455        Health Outcomes   

                                                  Measure Data_Value_Unit  \
0       Current lack of health insurance among adults ...               %   
1                  Arthritis among adults aged >=18 years               %   
2                  Arthritis among adults aged >=18 years               %   
3       Cholesterol screening among adults aged >=18 y...               %   
4                     Stroke among adults aged >=18 years               %   
...                                                   ...             ...   
188451  High cholesterol among adults aged >=18 years ...               %   
188452  Mental health not good for >=14 days among adu...               %   
188453       Current smoking among adults aged >=18 years               %   
188454  Fair or poor self-rated health status among ad...               %   
188455            Depression among adults aged >=18 years               %   

                Data_Value_Type  Data_Value  ...  Data_Value_Footnote  \
0              Crude prevalence        15.3  ...                  NaN   
1              Crude prevalence        34.8  ...                  NaN   
2              Crude prevalence        37.8  ...                  NaN   
3       Age-adjusted prevalence        86.6  ...                  NaN   
4              Crude prevalence         4.2  ...                  NaN   
...                         ...         ...  ...                  ...   
188451         Crude prevalence        34.6  ...                  NaN   
188452  Age-adjusted prevalence        13.9  ...                  NaN   
188453         Crude prevalence        18.1  ...                  NaN   
188454  Age-adjusted prevalence        18.8  ...                  NaN   
188455  Age-adjusted prevalence        20.3  ...                  NaN   

        Low_Confidence_Limit  High_Confidence_Limit  TotalPopulation  \
0                       15.0                   15.7      331,449,281   
1                       33.6                   36.1           55,411   
2                       36.5                   39.1           10,650   
3                       85.4                   87.7           48,959   
4                        3.8                    4.7           48,959   
...                      ...                    ...              ...   
188451                  33.7                   35.5          130,789   
188452                  12.8                   15.0           35,751   
188453                  15.2                   20.9           38,383   
188454                  16.3                   21.3          176,253   
188455                  19.1                   21.5          102,902   

       LocationID  CategoryID   MeasureId DataValueTypeID  \
0              59     PREVENT     ACCESS2          CrdPrv   
1            1033     HLTHOUT   ARTHRITIS

In [16]:
del county['LocationID'] # not helpful
del county['Short_Question_Text'] # repeat of measureID
del county['Category'] # repeat in categoryID
del county['Measure'] # measureID summarizes it

county['ConfidenceLimit'] = county[['Low_Confidence_Limit', 'High_Confidence_Limit']].apply(tuple, axis=1)
del county['Low_Confidence_Limit']
del county['High_Confidence_Limit']

del county['Data_Value_Unit']
del county['StateDesc']
del county['Data_Value_Footnote_Symbol']
del county['Data_Value_Footnote']
del county['Data_Value_Type']

county = county[county['Data_Value'].notna()]

county = county[county['LocationName'].notna()] # dropping 60 rows
county = county.rename(columns={'StateAbbr': 'State', 'Data_Value': 'DataValue', 'Short_Question_Text':'ShortQuestionText', 'MeasureId':'MeasureID', 'Geolocation':'GeoLocation'})

county.isnull().sum() # now left with no NA values

Year               0
State              0
LocationName       0
DataSource         0
DataValue          0
TotalPopulation    0
CategoryID         0
MeasureID          0
DataValueTypeID    0
GeoLocation        0
ConfidenceLimit    0
dtype: int64

In [17]:
county

Year State LocationName DataSource  DataValue TotalPopulation  \
1       2020    AL      Colbert      BRFSS       34.8          55,411   
2       2020    AL        Coosa      BRFSS       37.8          10,650   
3       2019    AL         Dale      BRFSS       86.6          48,959   
4       2020    AL         Dale      BRFSS        4.2          48,959   
5       2020    AL       DeKalb      BRFSS       36.4          71,658   
...      ...   ...          ...        ...        ...             ...   
188451  2019    WA       Skagit      BRFSS       34.6         130,789   
188452  2020    WI       Oneida      BRFSS       13.9          35,751   
188453  2020    WI       Oconto      BRFSS       18.1          38,383   
188454  2020    WV      Kanawha      BRFSS       18.8         176,253   
188455  2020    WI  Fond du Lac      BRFSS       20.3         102,902   

       CategoryID   MeasureID DataValueTypeID  \
1         HLTHOUT   ARTHRITIS          CrdPrv   
2         HLTHOUT   ARTHRITIS          CrdPrv   
3         PREVENT  CHOLSCREEN       AgeAdjPrv   
4         HLTHOUT      STROKE          CrdPrv   
5         HLTHOUT   ARTHRITIS          CrdPrv   
...           ...         ...             ...   
188451    HLTHOUT    HIGHCHOL          CrdPrv   
188452   HLTHSTAT       MHLTH       AgeAdjPrv   
188453    RISKBEH    CSMOKING          CrdPrv   
188454   HLTHSTAT       GHLTH       AgeAdjPrv   
188455    HLTHOUT  DEPRESSION       AgeAdjPrv   

                            GeoLocation ConfidenceLimit  
1         POINT (-87.8014569 34.703112)    (33.6, 36.1)  
2        POINT (-86.2434818 32.9314453)    (36.5, 39.1)  
3         POINT (-85.609476 31.4306536)    (85.4, 87.7)  
4         POINT (-85.609476 31.4306536)      (3.8, 4.7)  
5        POINT (-85.8040207 34.4609148)    (35.0, 37.8)  
...                                 ...             ...  
188451  POINT (-121.8157703 48.4932917)    (33.7, 35.5)  
188452   POINT (-89.5345333 45.7161762)    (12.8, 15.0)  
188453   POINT (-88.2065162 44.9965754)    (15.2, 20.9)  
188454   POINT (-81.5235156 38.3280675)    (16.3, 21.3)  
188455    POINT (-88.493284 43.7547219)    (19.1, 21.5)  

[188396 rows x 11 columns]

## Exporting df to csv